### DISCUSSION TITLE
Formatting IEnumerable<T> as table

### DISCUSSION THREAD

**Discussion by scottstephens** [dotnet/interactive#3081] [Q&A] [answered]
In the initial versions of this the default formatter for an `IEnumerable<T>` was a table display, with each row corresponding to one element of the `IEnumerable<T>`, and each column corresponding to a field or property of `T`. It was featured in the [preview 1 launch announcement](https://devblogs.microsoft.com/dotnet/net-core-with-juypter-notebooks-is-here-preview-1/#list-of-features) to display the `groupofheros` variable.

It seems this is no longer the default. Is there any way to do this in newer versions? It's pretty useful to be able to look at a collection that way.

**Comment by jonsequitur** [comment]
We're talking about bringing this back as a configurable behavior. The nested tables were very hard to use when the objects in the list were of different types, so we aligned it more to the watch-style UI in Visual Studio.

**Reply by jonsequitur** [reply, marked as answer]
One thing you can do is convert the objects to a `TabularDataResource` and then display that. Here's an example with the new formatter and then the `TabularDataResource` approach:

<img width="678" alt="image" src="https://github.com/dotnet/interactive/assets/547415/5ad65849-c490-43a4-b7d7-68ce06ea3f2b">

**Reply by scottstephens** [reply]
Great, thanks!

I think calling a method to trigger the table display is very usable, I don't feel the need for a config option to change the default, even though I almost always want table output. 

It could use a mention in https://github.com/dotnet/interactive/blob/main/docs/display-output-csharp.md, I'll send you a pull request for that.

It does seem important enough that a more concise name (.Table() or .DisplayTable() maybe) and/or inclusion in the namespace by default would be appropriate.

**Reply by jonsequitur** [reply]
Agreed. Something with clearer ergonomics and better discoverability would be better. I like the suggestion of `DisplayTable` because it will be more discoverable in IntelliSense.

**Reply by mlaily** [reply]
If I may add my two cents, I guess having a `displayTable` function for F# would be nice too, to complement the existing `display`!

**Comment by jonsequitur** [comment]
The `IEnumerable<T>.DisplayTable` extension method has been added to make this more straightforward. More details here: #3098.
